ot/lp/__init__.py

```diff
@@ -302,6 +304,9 @@ def emd(a, b, M, numItermax=100000, log=False, center_dual=True, numThreads=1, c
     ot.optim.cg : General regularized OT
     """
 
+    if np.isnan(M).any():
+        raise ValueError('The loss matrix should not contain NaN values.')
+
```

Comment:
Failing early here ensures that we do not segfault in the accelerated `emd_c` call. 

I did not look too deep into the `emd_c` implementation, but my assumption is that this check is somewhat pessimistic. Maybe it is possible to formulate problems for which we do not need to access a subset of values in the loss matrix (possibly due to the graph being disconnected). In that case we could support NaN values in some cases. @rflamary what is your opinion on this?

Reply:
if the graph is disconnected then the parts that are not used should have an infinite value (which is ha,ndled by the C++ solver). i'm OK with not handling naNs.